src/box/sql/resolve.c

```diff
@@ -981,6 +981,14 @@ sqlResolveOrderGroupBy(Parse * pParse,	/* Parsing context.  Leave error messages
 			resolveAlias(pEList, pItem->u.x.iOrderByCol - 1,
 				     pItem->pExpr, zType, 0);
 		}
+		enum field_type type = sql_expr_type(pItem->pExpr);
+		if (type == FIELD_TYPE_ARRAY || type == FIELD_TYPE_MAP ||
+		    type == FIELD_TYPE_INTERVAL) {
```

Comment:
Maybe you should better use `field_type1_contains_type2(FIELD_TYPE_SCALAR, type)`. To check if the type just isn't scalar. The result will be the same, but you won't hardcode specific types.

Reply:
Thank you for the suggestion! Fixed. However, we can't disallow ANY here for the time being, as it's possible that values of type ANY might be resolved later. Added a check for this.